src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -276,8 +292,13 @@ var ReactMultiChild = {
      */
     _updateChildren: function(nextNestedChildrenElements, transaction, context) {
       var prevChildren = this._renderedChildren;
+      var removedChildren = {};
```

Comment:
This is an unfortunate required allocation for every component. I believe that there is currently no allocation for leaves.